### PR TITLE
✅  Create plugin navigation utils

### PR DESCRIPTION
**Pull request by BeltranBulbarellaDD**
## Motivation

Some plugins tests are flaky the idea of this PR is to try and mitigate that. 

## Changes

Add `data-testid` to plugin tests apps so we don't rely on hardcoded text. 
Create `navigationUtils` for plugins with:
- clickAndWait
- navigate
- clickAndWaitForURL
- goHome

## Test instructions

Run the whole CI, specifically with `--repeat-each` or 
`yarn playwright test --config test/e2e/playwright.local.config.ts -g "plugin"` to run on all browsers. 

## Checklist

- [X] Tested locally
- [ ] Tested on staging
- [ ] Added unit tests for this change.
- [X] Added e2e/integration tests for this change.
- [ ] Updated documentation and/or relevant AGENTS.md file

### CODE REVIEW

File: test/e2e/scenario/browser-extensions/browserExtensions.scenario.ts [hunk 1]
```diff
@@ -92,7 +92,7 @@ test.describe('browser extensions', () => {
    */
   createTest('should not warn - edge case simulating NextJs with an extension that override `appendChild`')
     .withExtension(createExtension(path.join(BASE_PATH, 'appendChild-extension')))
-    .withRum()
+    .withRum({ sessionReplaySampleRate: 0 })
```
Comment (thomas-lebeau) [inline test/e2e/scenario/browser-extensions/browserExtensions.scenario.ts:95]: **❓ question:** ‏what is this for?

File: test/e2e/scenario/plugins/angularPlugin.scenario.ts [hunk 2]
```diff
@@ -38,26 +40,28 @@ test.describe('plugin: angular', () => {
     .withRum()
     .withApp(angularAppName)
     .run(async ({ page, flushEvents, intakeRegistry }) => {
-      await page.click('text=Go to Nested Route')
+      await clickAndWaitForURL(page, '[data-testid="go-to-nested-route"]', '**/parent/nested')
```
Comment (thomas-lebeau) [inline test/e2e/scenario/plugins/angularPlugin.scenario.ts:43]: **💭 thought:** ‏Do you need that? Playwright is [waiting for navigation](https://playwright.dev/docs/api/class-locator#locator-click:~:text=Wait%20for%20initiated%20navigations%20to%20either%20succeed%20or%20fail%2C%20unless%20noWaitAfter%20option%20is%20set.) ootb. is it not working?

File: test/e2e/scenario/browser-extensions/browserExtensions.scenario.ts (hunk 1 above)
Comment (BeltranBulbarellaDD) [reply test/e2e/scenario/browser-extensions/browserExtensions.scenario.ts:95]: Because at the end we assert `withBrowserLogs((logs) => expect(logs.length).toBe(0))` I had a failed test because of a log of the recorder. Since we are not using sessionReplay in this test, I though of removing it by doing this.

File: test/e2e/scenario/plugins/angularPlugin.scenario.ts (hunk 2 above)
Comment (BeltranBulbarellaDD) [reply test/e2e/scenario/plugins/angularPlugin.scenario.ts:43]: That's a good point. I didn't know it `Wait for initiated navigations to either succeed or fail`. So I was doing it for all tests, click and wait. I'm going to try to simply it by removing it and see if I see any flakyness.

File: test/e2e/scenario/browser-extensions/browserExtensions.scenario.ts (hunk 1 above)
Comment (thomas-lebeau) [reply test/e2e/scenario/browser-extensions/browserExtensions.scenario.ts:95]: session replay should not log anything, what was the log?

**Comment by BeltranBulbarellaDD** [comment]
Close until we find reason to believe we have serious flakyness in the CI for plugins
